src/views/EmployeePage.vue

```diff
@@ -117,7 +275,24 @@
             },
             mapEmployeeDepartments(departments) {
                 return departments.map(d => d.dept_name).join(', ');
-            }
+            },
+            saveChanges() {
+              let newEmployeeData = {...this.editedEmployee};
+              if (newEmployeeData.telephone === this.employee.telephone) {
+                delete newEmployeeData.telephone;
+              }
+              ApiService.editEmployee(this.employee.emp_no, newEmployeeData)
+                  .then(r => {
+                    if (r.emp_no === this.employee.emp_no) {
+                      this.employee = this.editedEmployee;
```

Comment:
this.employee is only used for 'setHeader'. Why we need 3 identical objects (this.editedEmployee, this.employee, newEmployeeData)?